### PR TITLE
feat(stark-ui): improve SessionUI pages to keep consistency

### PR DESCRIPTION
**Pull request by cnomes**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #718 

## What is the new behavior?
Session pages are styled through the same `stark-session-card` component.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

https://github.com/NationalBankBelgium/stark/pull/1167#issuecomment-469294490

### CODE REVIEW

**Comment by christophercr** [comment]
I'm wondering whether this is a breaking change or not due to the fact that the CSS involved to change the Logo at the header of the session pages will be different now. What do you think? @cnomes @SuperITMan

**Comment by cnomes** [comment]
> Small remarks. And also... can you add this option: `"extractCss": true,` under `projects.showcase.architect.build.options` to the `angular.json` of the Showcase and Starter?
> 
> This way the global CSS will be extracted to a separate CSS file instead of being a JS file which will make the styles load faster. So the "Loading" animation we have while the app is loading will be styled correctly without flickering and it will be possible to see the raw CSS files while debugging in the browser 😉

Update 👍

**Comment by cnomes** [comment]
> I'm wondering whether this is a breaking change or not due to the fact that the CSS involved to change the Logo at the header of the session pages will be different now (which is something that almost every app customizes). What do you think? @cnomes @SuperITMan

I agree that it is a breaking change, I had not considered that.

**Comment by cnomes** [comment]
@christophercr I added this commit msg:

feat(stark-ui): implement stark-session-card in stark session pages

ISSUES CLOSED: #718

BREAKING CHANGE: Refactor of Session Pages (login / logout / sessionExpired / preloading).

  - **stark-ui:** css on these components to change the default Stark logo will no longer work and should be refactored.
    For example adding this to `src/styles/styles.scss`
```SCSS
.stark-app-logo i {
  background-image: url("/assets/path-to-your-logo.png");
  background-size: contain;
}
```

File: packages/stark-ui/src/modules/session-ui/pages/login/login-page.component.spec.ts [hunk 1]
```diff
@@ -15,7 +19,8 @@ import {
 	MockStarkUserService
 } from "@nationalbankbelgium/stark-core/testing";
 import { TranslateModule } from "@ngx-translate/core";
-import { RawParams } from "@uirouter/core";
+import { StarkAppLogoModule } from "@nationalbankbelgium/stark-ui";
```
Comment (christophercr) [inline packages/stark-ui/src/modules/session-ui/pages/login/login-page.component.spec.ts]: This import should be relative ;)

File: packages/stark-ui/src/modules/session-ui/pages/session-expired/session-expired-page.component.spec.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+import { TranslateModule } from "@ngx-translate/core";
 import { CommonModule } from "@angular/common";
-import { STARK_APP_CONFIG, STARK_LOGGING_SERVICE, StarkApplicationConfig } from "@nationalbankbelgium/stark-core";
-import { MockStarkLoggingService } from "@nationalbankbelgium/stark-core/testing";
+import { MatCardModule } from "@angular/material/card";
 import { MatButtonModule } from "@angular/material/button";
-import { TranslateModule } from "@ngx-translate/core";
+import { STARK_APP_CONFIG, STARK_LOGGING_SERVICE, STARK_ROUTING_SERVICE,StarkApplicationConfig } from "@nationalbankbelgium/stark-core";
+import { MockStarkLoggingService,MockStarkRoutingService } from "@nationalbankbelgium/stark-core/testing";
+import { StarkAppLogoModule } from "@nationalbankbelgium/stark-ui";
```
Comment (christophercr) [inline packages/stark-ui/src/modules/session-ui/pages/session-expired/session-expired-page.component.spec.ts]: This import should be relative ;)

File: packages/stark-ui/src/modules/session-ui/components/session-card/session-card.component.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<mat-card>
```
Comment (christophercr) [inline packages/stark-ui/src/modules/session-ui/components/session-card/session-card.component.html]: Before, in every session page we had `role="alertdialog" aria-busy="true" aria-live="assertive"`. Is this not relevant anymore? Maybe it is still relevant right?

File: packages/stark-ui/src/modules/session-ui/pages/preloading/preloading-page.component.spec.ts [hunk 4]
```diff
@@ -14,9 +18,8 @@ import {
 	MockStarkSessionService,
 	MockStarkUserService
 } from "@nationalbankbelgium/stark-core/testing";
-import { TranslateModule } from "@ngx-translate/core";
-import { MatButtonModule } from "@angular/material/button";
-import { of, throwError } from "rxjs";
+import { StarkAppLogoModule } from "@nationalbankbelgium/stark-ui";
```
Comment (christophercr) [inline packages/stark-ui/src/modules/session-ui/pages/preloading/preloading-page.component.spec.ts]: This import should be relative ;)

File: packages/stark-ui/src/modules/session-ui/pages/session-logout/session-logout-page.component.spec.ts [hunk 5]
```diff
@@ -1,9 +1,12 @@
...
 import { async, ComponentFixture, TestBed } from "@angular/core/testing";
-import { CommonModule } from "@angular/common";
-import { STARK_APP_CONFIG, STARK_LOGGING_SERVICE, StarkApplicationConfig } from "@nationalbankbelgium/stark-core";
-import { MockStarkLoggingService } from "@nationalbankbelgium/stark-core/testing";
 import { TranslateModule } from "@ngx-translate/core";
+import { CommonModule } from "@angular/common";
+import { MatCardModule } from "@angular/material/card";
+import { STARK_APP_CONFIG, STARK_LOGGING_SERVICE, STARK_ROUTING_SERVICE, StarkApplicationConfig } from "@nationalbankbelgium/stark-core";
+import { MockStarkLoggingService, MockStarkRoutingService } from "@nationalbankbelgium/stark-core/testing";
+import { StarkAppLogoModule } from "@nationalbankbelgium/stark-ui";
```
Comment (christophercr) [inline packages/stark-ui/src/modules/session-ui/pages/session-logout/session-logout-page.component.spec.ts]: This import should be relative ;)

File: packages/stark-ui/src/modules/session-ui/pages/login/login-page.component.spec.ts (hunk 1 above)
Comment (cnomes) [reply packages/stark-ui/src/modules/session-ui/pages/login/login-page.component.spec.ts]: I thought that is why we have `"@nationalbankbelgium/stark-ui": ["."]` in the tslint.json? 😕

Comment (christophercr) [reply packages/stark-ui/src/modules/session-ui/pages/login/login-page.component.spec.ts]: True, you're right... my bad :(

File: packages/stark-ui/src/modules/session-ui/components/session-card/session-card.component.html (hunk 3 above)
Comment (cnomes) [reply packages/stark-ui/src/modules/session-ui/components/session-card/session-card.component.html]: I added back some ARIA attributes that seemed relevant.

File: packages/stark-ui/src/modules/session-ui/components/session-card/session-card.component.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+	host: {
+		class: componentName
+	}
+})
+export class StarkSessionCardComponent {
+	/**
+	 * The title shown in the header of the card
+	 */
+	@Input()
+	public title: string;
```
Comment (christophercr) [inline packages/stark-ui/src/modules/session-ui/components/session-card/session-card.component.ts]: As we discussed, we should rename this input to someting else cause `title` is an standard HTML attribute that shows a tooltip on the element. In this case it is showing a tooltip in the Card :s

**Pull request merged**